### PR TITLE
Fix #3103 Only last response is generated into Endpoint code

### LINKED ISSUE #3103: Only last response is generated into Endpoint code
If an OpenAPI spec operation defines multiple responses, the generated code only acknowledges the last response in the list. It won't contain multiple `.outError`/ `.out` statements.

```scala
test("Generate all responses") {
        val oapi =
          OpenAPI(
            openapi = "3.0.0",
            info = OpenAPI.Info(
              title = "XXX",
              description = None,
              termsOfService = None,
              contact = None,
              license = None,
              version = "1.0.0",
            ),
            paths = ListMap(
              OpenAPI.Path(name = "/api/a/b") -> OpenAPI.PathItem(
                ref = None,
                summary = None,
                description = None,
                get = None,
                put = None,
                post = Some(
                  OpenAPI.Operation(
                    summary = None,
                    description = None,
                    externalDocs = None,
                    operationId = None,
                    requestBody = None,
                    responses = Map(
                      OpenAPI.StatusOrDefault.StatusValue(status = Status.Ok)  ->
                        OpenAPI.ReferenceOr.Or(value = OpenAPI.Response()),
                      OpenAPI.StatusOrDefault.StatusValue(Status.BadRequest)   ->
                        OpenAPI.ReferenceOr.Or(OpenAPI.Response()),
                      OpenAPI.StatusOrDefault.StatusValue(Status.Unauthorized) ->
                        OpenAPI.ReferenceOr.Or(OpenAPI.Response()),
                    ),
                  ),
                ),
                delete = None,
                options = None,
                head = None,
                patch = None,
                trace = None,
              ),
            ),
            components = None,
            externalDocs = None,
          )

        val maybeEndpointCode =
          EndpointGen
            .fromOpenAPI(oapi, Config.default)
            .files
            .flatMap(_.objects)
            .flatMap(_.endpoints)
            .collectFirst {
              case (field, code) if field.name == "post" => code
            }

        assertTrue(
          maybeEndpointCode.is(_.some).outCodes.length == 1 &&
            maybeEndpointCode.is(_.some).errorsCode.length == 2,
        )
      },
```

### PR DESCRIPTION
**Pull request by nafg**
/claim #3103

### CODE REVIEW

**Comment by nafg** [comment]
The issue was that the code was doing `.collect { ... => ... (a, b) }` on a Map, which returns another Map. But `a` is just the imports needed and may not be unique. Inserting a `.toSeq` fixed it.

**Review by 987Nabil — APPROVED** [review]

File: zio-http-gen/src/test/scala/zio/http/gen/scala/CodeGenSpec.scala [hunk 1]
```diff
@@ -975,5 +976,67 @@ object CodeGenSpec extends ZIOSpecDefault {
...
+            info = OpenAPI.Info(
+              title = "XXX",
+              description = None,
+              termsOfService = None,
+              contact = None,
+              license = None,
+              version = "1.0.0",
+            ),
+            paths = ListMap(
+              OpenAPI.Path(name = "/api/a/b") -> OpenAPI.PathItem(
```
Comment (987Nabil) [inline zio-http-gen/src/test/scala/zio/http/gen/scala/CodeGenSpec.scala]: Path has no public constructor but only a smart constructor. Calling .get should be okay here

Comment (nafg) [reply zio-http-gen/src/test/scala/zio/http/gen/scala/CodeGenSpec.scala]: Oh that compiles on `Scala213`, I guess you're not using `-Xsource:3`. I took a slightly different approach to fixing it.

**Comment by nafg** [comment]
Oh scala 2.12... :(

**Pull request merged**